Chaincase.SSB/JsInteropConfigProvider.cs

```diff
@@ -0,0 +1,50 @@
+using System.Text.Json;
+using System.Threading.Tasks;
+using Microsoft.JSInterop;
+
+namespace Chaincase.SSB
+{
+	public class JsInteropConfigProvider
+	{
+		private readonly IJSRuntime _jsRuntime;
+
+		public JsInteropConfigProvider(IJSRuntime jsRuntime)
+		{
+			_jsRuntime = jsRuntime;
+		}
+
+		public virtual async Task<T> Get<T>(string key)
+		{
+			var lsRes = await GetRaw(key);
+
+			if (lsRes is null)
+			{
+				return default(T);
+			}
+
+			return JsonSerializer.Deserialize<T>(lsRes);
+		}
+
+		protected async Task<string> GetRaw(string key)
+		{
+			return await _jsRuntime.InvokeAsync<string>("localStorage.getItem", key);
+		}
+
+		public virtual async Task Set<T>(string key, T value)
+		{
+			if (value.Equals(default(T)))
+			{
+				await _jsRuntime.InvokeVoidAsync("localStorage.removeItem", key);
```

Comment:
‼️ should be `localStorage.setItem` a la `await _jsRuntime.InvokeVoidAsync("localStorage.setItem", key);` ❔

Reply:
default(T) == unset  as getting an unset config item == default(T)

Reply:
Still unclear. Logic now reads ~ if default value set, rermove it [and do nothing] else set key: value.

Reply:
@Kukks gonna merge this but this is still confusing me

Reply:
```cs
 public class Bukkake
    {
        public string X { get; set; }
        public int Y { get; set; }

        private static Bukkake Item;

        public static Bukkake Load()
        {
            //null is the default for this obj
            return Item; // ?? default(Bukkake);
        }

        public static void Save(Bukkake bukkake)
        {
            if (bukkake == default(Bukkake))
            {
                //remove, we do not need to store a null value in the storage.
                Item = null;
            }
            else
            {
                Item = bukkake;
            }
        }
    }
```